### ISSUE TITLE
Renaming script for parallel builds not working

### ISSUE THREAD

**Issue by ju-w** [ankidroid/Anki-Android#4777]
When I try to invoke the renaming  script with `./tools/parallel-package-name.sh 'com.ichi2.anki.a' 'An
kiDroid.A'` I get the following errors:

```
./tools/parallel-package-name.sh: line 4: $'\r': command not found
./tools/parallel-package-name.sh: line 8: $'\r': command not found
./tools/parallel-package-name.sh: line 12: $'\r': command not found
: No such file or directory
/name="app_name">AnkiDroid.A/g: No such file or directory
: No such file or directoryrc/main/
/applicationId "com.ichi2.anki.a/g: No such file or directory
: No such file or directoryuild.gradle
/android:authorities="com.ichi2.anki.a/g: No such file or directory
: No such file or directoryrc/main/
.permission/permission android:name="com.ichi2.anki.a.permission/g: No such file or directory
: No such file or directoryrc/main/
find: missing argument to `-exec'
```

Caused by `CRLF` line endings.

However, when I change the endings to `LF` I get the errors:

```
/name="app_name">AnkiDroid.A/g: No such file or directory
/applicationId "com.ichi2.anki.a/g: No such file or directory
/android:authorities="com.ichi2.anki.a/g: No such file or directory
.permission/permission android:name="com.ichi2.anki.a.permission/g: No such file or directory
"/android:targetPackage="com.ichi2.anki.a"/g: No such file or directory
"/android:targetPackage="com.ichi2.anki.a"/g: No such file or directory
"/android:targetPackage="com.ichi2.anki.a"/g: No such file or directory
"/android:targetPackage="com.ichi2.anki.a"/g: No such file or directory
"/android:targetPackage="com.ichi2.anki.a"/g: No such file or directory
"/android:targetPackage="com.ichi2.anki.a"/g: No such file or directory
"/android:targetPackage="com.ichi2.anki.a"/g: No such file or directory
"/android:targetPackage="com.ichi2.anki.a"/g: No such file or directory
"/android:targetPackage="com.ichi2.anki.a"/g: No such file or directory
"/android:targetPackage="com.ichi2.anki.a"/g: No such file or directory
"/android:targetPackage="com.ichi2.anki.a"/g: No such file or directory
"/android:targetPackage="com.ichi2.anki.a"/g: No such file or directory
"/android:targetPackage="com.ichi2.anki.a"/g: No such file or directory
"/android:targetPackage="com.ichi2.anki.a"/g: No such file or directory
"/android:targetPackage="com.ichi2.anki.a"/g: No such file or directory
```

It would be great if somone could look into this.

**Comment by mikehardy** [comment]
Oh my, that's not good. That's my script and for what it's worth it was built and tested on a Mac OS X rig. I know that the OS X sed is slightly different than others, and it looks like that may be the culprit. Some questions if you can answer -
1- what's your development environment for this? (with sed -v information if possible)
2- I'm sure you are, but can you confirm you are using the most up to date code?

As two possible workarounds for you if this is urgent, you can download the builds I created and linked to if you like, or you can also use App Cloner to make at least 1 clone for free (more than 1 clone you have to buy it) https://play.google.com/store/apps/details?id=com.applisto.appcloner&hl=en - just make sure you change the storage directory

**Comment by ju-w** [comment]
Hi,

1. This is on Ubuntu Linux with sed (GNU sed) 4.2.2.
2. Yes, up to date indeed (e6eb737a400a34ce8e9d2268e0b1566d7a2119d9).

Thank you for your input on workarounds, I'll try to figure something out. If it's not too complicated it would still be nice if it could be fixed.

**Comment by mikehardy** [comment]
I will absolutely fix it but it will be later this evening (and likely just on my branch until merged) - I've got a few ubuntu VMs around as well so should be able to reproduce easily. Thanks for the extra information

**Comment by ju-w** [comment]
Sounds great, I appreciate it!

**Comment by mikehardy** [comment]
I believe my commit (and associated pull request) fixes this.
tl;dr is that yes it was sed, the '-i' in-place editing flag isn't POSIX so wasn't cross-platform. I converted it to 'perl -pi -e' style, seems to work on OSX and Linux now. 

Thanks for noticing this and taking the time to give me more information so I could fix it.
